drv/transceivers-server/src/udp.rs

```diff
@@ -0,0 +1,503 @@
+                    }
+                    Err(e) => {
+                        // At this point, deserialization has failed, so we
+                        // can't handle the packet.  We'll attempt to
+                        // deserialize *just the header* (which should never
+                        // change), in the hopes of logging a more detailed
+                        // error message about a version mismatch.
+                        ringbuf_entry!(Trace::DeserializeError(e));
+                        match hubpack::deserialize::<Header>(rx_data_buf) {
+                            Ok((header, _)) => {
+                                ringbuf_entry!(Trace::WrongVersion(
+                                    header.version
+                                ));
+                            }
+                            Err(e) => {
+                                ringbuf_entry!(Trace::DeserializeHeaderError(
+                                    e
+                                ));
+                            }
+                        }
+                        None
+                    }
+                };
+
+                if let Some(out_len) = out_len {
+                    meta.size = out_len;
+                    if let Err(e) = self.net.send_packet(
+                        SOCKET,
+                        meta,
+                        &tx_data_buf[..meta.size as usize],
+                    ) {
+                        // We'll drop packets if the outgoing queue is full;
+                        // the host is responsible for retrying.
+                        //
+                        // Other errors are unexpected and panic.
+                        ringbuf_entry!(Trace::SendError(e));
+                        match e {
+                            SendError::QueueFull => (),
+                            SendError::Other
+                            | SendError::NotYours
+                            | SendError::InvalidVLan => panic!(),
+                        }
+                    }
+                }
+            }
+            Err(RecvError::QueueEmpty) => {
+                // Our incoming queue is empty. Wait for more packets
+                // in dispatch_n, back in the main loop.
+            }
+            Err(RecvError::NotYours | RecvError::Other) => panic!(),
+        }
+    }
+
+    /// Handles a single message from the host with supplementary data in `data`
+    ///
+    /// Returns a response and a `usize` indicating how much was written to the
+    /// `out` buffer.
+    fn handle_message(
+        &mut self,
+        msg: Message,
+        data: &[u8],
+        tx_data_buf: &mut [u8],
+    ) -> Option<u32> {
+        // If the version is mismatched, then we can't trust the deserialization
+        // (even though it nominally succeeded); don't reply.
+        if msg.header.version != 1 {
+            ringbuf_entry!(Trace::WrongVersion(msg.header.version));
+            return None;
```

Comment:
Similar comment as above - should this send a protocol error message back to the host?

Reply:
We decided to silently drop cases where the version doesn't match, since it means that we can't trust much of anything.  Also, if we did reply, our serialized message _back_ would necessarily be encoded with **our** version, meaning the host would see a version mismatch and try to reply with a protocol error, etc.